### DISCUSSION TITLE
A class to check for marking returning marker name.

### DISCUSSION THREAD

**Discussion by ancelin-charpentier** [cwapi3d/cwapi3dpython#221] [Q&A] [answered]
Hey there, @Brunner246 
There are few APIs that I would like to create which would need the ability to check if an element in a model has marking onto itself.
By marking I am referring to what this menu offers:
![image](https://github.com/user-attachments/assets/833c861b-5d83-4afb-8716-dd38c4a4f1a7)

Is there already ways to do the following:
- Identify if an element **has markings** due to conditions given in the "Marking setting" menu? and if yes, return either the marking condition or the markers name?
-  Identify if an element **is a marker** due to conditions given in the "Marking setting" menu? and if yes, return either the marking condition or the marked parts name?

Thanks for your help.

**Comment by ancelin-charpentier** [comment]
Hey @Brunner246 ,
Sorry for the harassment but just checking if you have seen the above? There are several case that the factory raised to me recently which would be solved by using this option in an API.
Have a nice day

**Reply by Brunner246** [reply, marked as answer]
Hi @ancelin-charpentier 
We will check if we can add some functionality for this use-cases.

**Comment by ancelin-charpentier** [comment]
In addition to the query above, it would very practical to get a square meterage of the "Marked area" if any.
